stan/math/rev/core/init_chainablestack.hpp

```diff
@@ -23,7 +23,7 @@ namespace math {
  * Refer to https://software.intel.com/en-us/node/506314 for details
  * on the observer concept.
  */
-class ad_tape_observer : public tbb::task_scheduler_observer {
+class ad_tape_observer final : public tbb::task_scheduler_observer {
   using stack_ptr = std::unique_ptr<ChainableStack>;
```

Comment:
Can get rid of this line now.

Reply:
The final? That's the only thing that changed. I can remove but it doesn't make a difference either way. In some places the compiler knowing a class is `final` can let it do some extra optimizations

Reply:
The stack_ptr, sry.

Reply:
Oh, so I'm going to change the guts of this function in a separate PR. There's some ways we can make that make a bit more efficient. I don't really wanna mess with it in this PR